src/gt4py/next/common.py

```diff
@@ -135,36 +135,128 @@ def __and__(self, other: Set[Any]) -> UnitRange:
         else:
             raise NotImplementedError("Can only find the intersection between UnitRange instances.")
 
+    def __str__(self) -> str:
+        return f"({self.start}:{self.stop})"
+
+
+def unit_range(r: UnitRangeLike) -> UnitRange:
+    assert is_unit_range_like(r)
+    if isinstance(r, UnitRange):
+        return r
+    if isinstance(r, range) and r.step == 1:
+        return UnitRange(r.start, r.stop)
+    if isinstance(r, tuple) and isinstance(r[0], int) and isinstance(r[1], int):
+        return UnitRange(r[0], r[1])
+    raise ValueError(f"`{r}` is not `UnitRangeLike`.")
 
-DomainRange: TypeAlias = UnitRange | int
+
+IntIndex: TypeAlias = int | np.integer
+DomainRange: TypeAlias = UnitRange | IntIndex
 NamedRange: TypeAlias = tuple[Dimension, UnitRange]
-NamedIndex: TypeAlias = tuple[Dimension, int]
+NamedIndex: TypeAlias = tuple[Dimension, IntIndex]
+AnyIndex: TypeAlias = IntIndex | NamedRange | NamedIndex | slice | EllipsisType
 DomainSlice: TypeAlias = Sequence[NamedRange | NamedIndex]
-FieldSlice: TypeAlias = (
-    DomainSlice
-    | tuple[slice | int | EllipsisType, ...]
-    | slice
-    | int
-    | EllipsisType
-    | NamedRange
-    | NamedIndex
+BufferSlice: TypeAlias = tuple[slice | IntIndex | EllipsisType, ...]
+FieldSlice: TypeAlias = DomainSlice | BufferSlice | AnyIndex
+UnitRangeLike: TypeAlias = UnitRange | range | tuple[int, int]
+DomainLike: TypeAlias = (
+    Sequence[tuple[Dimension, UnitRangeLike]] | Mapping[Dimension, UnitRangeLike]
 )
 
 
-@dataclasses.dataclass(frozen=True)
+def is_int_index(p: Any) -> TypeGuard[IntIndex]:
+    return isinstance(p, (int, np.integer))
```

Comment:
It is because np.integer is not a simple type, I get
```
src/gt4py/next/common.py:186:13: error: Parameterized generics cannot be used with class or instance checks  [misc]
src/gt4py/next/common.py:186:30: error: Argument 2 to "isinstance" has incompatible type "<typing special form>"; expected "_ClassInfo"  [arg-type]
```
Probably we should not use np.integer, but `core_defs.IntegralScalar`?

Reply:
but it has the same problem

Reply:
resolved as discussed